scheduler/src/instance_listener.rs

```diff
@@ -0,0 +1,44 @@
+use log::debug;
+use tokio::sync::mpsc;
+use tokio_stream::wrappers::ReceiverStream;
+use tonic::{Request, Response, Status};
+
+use proto::scheduler::{
+    instance_service_server::InstanceService, Instance, InstanceIdentifier, InstanceStatus,
+};
+
+#[derive(Debug, Default)]
+pub struct InstanceListener {}
+
+#[tonic::async_trait]
+impl InstanceService for InstanceListener {
+    async fn create(
+        &self,
+        request: Request<Instance>,
+    ) -> Result<Response<Self::CreateStream>, Status> {
+        debug!("{:?}", request);
+
+        let (tx, rx) = mpsc::channel(4);
```

Comment:
I suppose the tx half will be used in upcoming PRs?

Reply:
Yes, I've pushed more commits for you to understand our structure.